pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -185,12 +199,20 @@ func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
 		return apis.ErrInvalidValue(err.Error(), "spec.tasks.params.value")
 	}
 
+	if err := ValidateTasksAndFinallySection(ps); err != nil {
+		return err
+	}
+
+	if err := ValidateFinalTasks(ps.Finally); err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // validatePipelineTasks ensures that pipeline tasks has unique label, pipeline tasks has specified one of
 // taskRef or taskSpec, and in case of a pipeline task with taskRef, it has a reference to a valid task (task name)
-func validatePipelineTasks(ctx context.Context, tasks []PipelineTask) *apis.FieldError {
+func validatePipelineTasks(ctx context.Context, tasks []PipelineTask, finalTasks []PipelineTask) *apis.FieldError {
```

Comment:
not sure i am following, at this point, finalTasks are still separate from pipelineTasks, finalTasks is a list of `PipelineTask` but here, tasks represents tasks under `tasks` section and finalTasks represents tasks under `finally` section

Reply:
I made an example to show what im trying to say: https://play.golang.org/p/C2USaY-jimW (which was probably overkill, i think im just excited to write some code XD)


basically im wondering if you can keep the function as-is:

```go
func validatePipelineTasks(ctx context.Context, tasks []PipelineTask)
```

And then instead of calling it like this:

```go
validatePipelineTasks(ctx, ps.Tasks, ps.Finally)
```

You can do this:


And then instead of calling it like this:

```go
validatePipelineTasks(ctx, append(ps.Tasks, ps.Finally...))
```

Now I realize we'd lose the field error tho 🤔  i feel like it might be worth it! maybe instead of passing "spec.tasks" or "spec.finally" in for the field name we could use something generic like "task.name"